business-networks-workflows/src/main/kotlin/net/corda/bn/flows/UpdateCordaIdentityFlow.kt

```diff
@@ -0,0 +1,274 @@
+        private val notary: Party? = null
+) : MembershipManagementFlow<SignedTransaction>() {
+
+    @Suspendable
+    override fun call(): SignedTransaction {
+        auditLogger.info("$ourIdentity started updating Corda identity of member with $membershipId membership ID")
+
+        val membership = getMembership(membershipId)
+        val name = membership.state.data.identity.cordaIdentity.name
+        val newIdentity = serviceHub.identityService.wellKnownPartyFromX500Name(name)
+                ?: throw FlowException("Party with $name X500 name doesn't exist")
+
+        // check whether party is authorised to initiate flow
+        val networkId = membership.state.data.networkId
+        val flowName = javaClass.name
+        getMembership(networkId, ourIdentity).apply {
+            if (!state.data.isActive()) {
+                throw IllegalMembershipStatusException("Membership owned by $ourIdentity is not active")
+            }
+            if (!state.data.canModifyGroups()) {
+                throw MembershipAuthorisationException("$ourIdentity is not authorised to run $flowName")
+            }
+        }
+
+        val updatedParticipantsList = membership.state.data.participants.map() {
+            if (it.owningKey == membership.state.data.identity.cordaIdentity.owningKey) {
+                newIdentity
+            } else it
+        }
+        val outputMembership = membership.state.data.run {
+            copy(identity = identity.copy(cordaIdentity = newIdentity), modified = serviceHub.clock.instant(),
+                    participants = updatedParticipantsList)
+        }
+
+        // fetch signers
+        val authorisedMemberships = getMembersAuthorisedToModifyMembership(networkId)
+        val signers = authorisedMemberships.filter {
+            it.state.data.isActive() && it.state != membership.state // Member's new identity is added at the end, avoid duplication
+        }.map {
+            it.state.data.identity.cordaIdentity
+        }.toSet() + newIdentity
+
+        // building transaction
+        val requiredSigners = signers.map { it.owningKey }
+        val builder = TransactionBuilder(notary ?: serviceHub.networkMapCache.notaryIdentities.first())
+                .addInputState(membership)
+                .addOutputState(outputMembership)
+                .addCommand(MembershipContract.Commands.ModifyCordaIdentity(requiredSigners), requiredSigners)
+        builder.verify(serviceHub)
+
+        // collect signatures and finalise transaction
+        val observerSessions = (outputMembership.participants - ourIdentity).map { initiateFlow(it) }
+        val finalisedTransaction = collectSignaturesAndFinaliseTransaction(builder, observerSessions, signers.toList())
+
+
+        // update memberships which reference affected member
+        subFlow(UpdateImpactedMembershipsFlow(membership, notary))
+
+        // update groups
+        subFlow(UpdateGroupMembersIdentityFlow(membership, notary))
```

Comment:
nit: Put method declarations in the order they are called ?

Reply:
That's a bit too much. Calling order can always change.